src/lib/y2partitioner/sequences/edit_blk_device.rb

```diff
@@ -49,20 +47,46 @@ def wizard_next_back(&block)
         Yast::Wizard.CloseDialog
       end
 
-      def format_mount
-        @dialog ||= Dialogs::FormatAndMount.new(@partition)
+      def format_options
+        @format_dialog ||= Dialogs::FormatAndMount.new(@options)
 
-        @dialog.run
+        @format_dialog.run
       end
 
       def password
-        if @partition.encryption
-          ret = Dialogs::EncryptPassword.new(@partition).run
+        return :next unless @options.encrypt
+        @encrypt_dialog ||= Dialogs::EncryptPassword.new(@options)
+
+        @encrypt_dialog.run
+      end
 
-          ret == :next ? :finish : ret
+      def format_mount
+        @partition.id = @options.partition_id
+        @partition.remove_descendants if @options.encrypt || @options.format
+
+        if @options.encrypt
+          @partition = @partition.create_encryption(dm_name_for(@partition))
+        end
+
+        @partition.create_filesystem(@options.filesystem_type) if @options.format
+
+        if @options.mount
+          @partition.filesystem.mount_point = @options.mount_point
+          @partition.filesystem.mount_by = @options.mount_by
+          @partition.filesystem.label = @options.label
+          @partition.filesystem.fstab_options = @options.fstab_options
         else
-          :finish
+          @partition.filesystem.mount_point = ""
         end
```

Comment:
Yep, I prefer to have duplication at the begging, which shows clearly that this should be moved to a common place like a Formatter class or just a method in FormatMountOptions